app/keys_loading/private_key_loading.py

```diff
@@ -0,0 +1,41 @@
+from hashlib import sha256
+from typing import Optional
+
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from .singleton_meta import SingletonMeta
+
+
+class PrivateKey(metaclass=SingletonMeta):
+    """
+    A class for managing the private key.
+    Only one instance of the class can exist throughout the application.
+    """
+
+    def __init__(self):
+        self.__private_key = None
+
+    def load_private_key(self, private_key_path: str, pin: str) -> None:
+        """
+        Loads the private key from the specified path and decrypts it with the specified PIN.
+        Uses SHA-256 to hash the PIN before usage.
+        :param:
+            private_key_path: The path to the private key file.
+            pin: A PIN to decrypt the private key.
+        """
+        hashed_pin = sha256(pin.encode()).digest()
+        with open(private_key_path, "rb") as f:
+            try:
+                self.__private_key = serialization.load_pem_private_key(
+                    f.read(), password=hashed_pin
+                )
+                print("private key was loaded")
+            except ValueError:
+                print("Invalid password")
```

Comment:
If you want to log some information for future debugging purposes or whatever it is better to use logging module from python rather that print statements